static/src/javascripts/projects/common/modules/identity/api.ts

```diff
@@ -0,0 +1,431 @@
+	const user = getUserFromCookie();
+
+	if (user) {
+		const endpoint = `${idApiRoot}/useremails/${user.id}`;
+		const request = fetch(endpoint, {
+			mode: 'cors',
+			credentials: 'include',
+		}).then((resp) => resp.json());
+
+		return request;
+	}
+
+	return Promise.resolve(null);
+};
+
+export const sendValidationEmail = (): unknown => {
+	const defaultReturnEndpoint = '/email-prefs';
+	const endpoint = `${idApiRoot}/user/send-validation-email`;
+
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- it’s okay
+	const returnUrl = getUrlVars().returnUrl
+		? // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- it’s okay
+		  decodeURIComponent(getUrlVars()?.returnUrl)
+		: profileRoot + defaultReturnEndpoint;
+
+	const data = new URLSearchParams();
+	data.append('method', 'post');
+	data.append('returnUrl', returnUrl);
+
+	const request = fetch(endpoint, {
+		mode: 'cors',
+		credentials: 'include',
+		body: data.toString(),
+	});
+
+	return request;
+};
+
+export const updateUsername = (username: string): unknown => {
+	const endpoint = `${idApiRoot}/user/me`;
+	const data = {
+		publicFields: {
+			username,
+			displayName: username,
+		},
+	};
+	const request = fetch(endpoint, {
+		mode: 'cors',
+		method: 'POST',
+		body: JSON.stringify(data),
+		credentials: 'include',
+	});
+
+	return request;
+};
+
+export const getAllConsents = (): Promise<unknown> => {
+	const endpoint = '/consents';
+	const url = idApiRoot + endpoint;
+	return fetchJson(url, {
+		mode: 'cors',
+		method: 'GET',
+		headers: { Accept: 'application/json' },
+	});
+};
+
+export const getAllNewsletters = (): Promise<unknown> => {
+	const endpoint = '/newsletters';
+	const url = idApiRoot + endpoint;
+	return fetchJson(url, {
+		mode: 'cors',
+		method: 'GET',
+		headers: { Accept: 'application/json' },
+	});
+};
+
+export const getSubscribedNewsletters = (): Promise<string[]> => {
+	const endpoint = '/users/me/newsletters';
+	const url = idApiRoot + endpoint;
+
+	type Subscriptions = {
+		listId: string;
+	};
+
+	type NewslettersResponse =
+		| {
+				result?: {
+					globalSubscriptionStatus?: string;
+					htmlPreference?: string;
+					subscriptions?: Subscriptions[];
+					status?: 'ok' | string;
+				};
+		  }
+		| undefined;
+
+	return fetchJson(url, {
+		mode: 'cors',
+		method: 'GET',
+		headers: { Accept: 'application/json' },
+		credentials: 'include',
+	})
+		.then((json: NewslettersResponse) => {
+			if (json?.result?.subscriptions) {
+				return json.result.subscriptions.map((sub) => sub.listId);
+			}
+			return [];
+		})
+		.catch(() => []);
+};
+
+export const setConsent = (consents: SettableConsent): Promise<void> =>
+	fetch(`${idApiRoot}/users/me/consents`, {
+		method: 'PATCH',
+		credentials: 'include',
+		mode: 'cors',
+		body: JSON.stringify(consents),
+	}).then((resp) => {
+		if (resp.ok) return Promise.resolve();
+		return Promise.reject();
+	});
+
+export const ajaxSignIn = (
```

Comment:
I think Identity has deprecated `smartlock / browser credentials` support https://github.com/guardian/identity-frontend/pull/607

Reply:
I’m afraid I can’t see that repo, but happy to remove this bit of code.